.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,50 @@
+---
+## Workflow to lint and test
+name: pull-request
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    env:
+      JENKINS_URL: https://beats-ci.elastic.co/
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
https://github.com/elastic/apm-pipeline-library/pull/1924 could be useful in addition to https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#testing-out-your-action-in-a-workflow